components/core/src/clp/ffi/ir_stream/decoding_methods.cpp

```diff
@@ -468,33 +470,69 @@ IRErrorCode deserialize_preamble(
     return IRErrorCode_Success;
 }
 
-IRProtocolErrorCode validate_protocol_version(std::string_view protocol_version) {
-    if ("v0.0.0" == protocol_version) {
-        // This version is hardcoded to support the oldest IR protocol version. When this version is
-        // no longer supported, this branch should be removed.
-        return IRProtocolErrorCode_Supported;
+auto validate_protocol_version(std::string_view protocol_version) -> IRProtocolErrorCode {
+    // These versions are hardcoded to support the IR protocol version that predates the key-value
+    // pair IR format.
+    constexpr std::array<std::string_view, 3> cBackwardCompatibleVersions{
```

Comment:
Do we want to include any backward compatible beta versions here?

Reply:
For old IR format no.
For key-value IR format, we do support `0.1.0-beta.1`. I added a util to strip the pre-release and only compare the version core.